packages/network-controller/src/NetworkController.ts

```diff
@@ -332,34 +333,31 @@ export class NetworkController extends BaseControllerV2<
     );
   }
 
-  #setupStandardProvider(
-    rpcTarget: string,
-    chainId?: string,
-    ticker?: string,
-    nickname?: string,
-  ) {
-    const config = {
-      chainId,
-      engineParams: { pollingInterval: 12000 },
-      nickname,
+  #setupStandardProvider(rpcTarget: string, chainId: Hex) {
+    const { provider, blockTracker } = createNetworkClient({
       rpcUrl: rpcTarget,
-      ticker,
-    };
-    this.#updateProvider(createMetamaskProvider(config));
+      chainId,
+      type: NetworkClientType.Custom,
+    });
+
+    this.#updateProvider(provider, blockTracker);
   }
 
-  #updateProvider(provider: Provider) {
+  #updateProvider(
+    provider: SafeEventEmitterProvider,
+    blockTracker: PollingBlockTracker,
+  ) {
     this.#safelyStopProvider(this.#provider);
     this.#setProviderAndBlockTracker({
       provider,
-      blockTracker: provider._blockTracker,
+      blockTracker,
     });
     this.#registerProvider();
   }
 
   #safelyStopProvider(provider: Provider | undefined) {
     setTimeout(() => {
-      provider?.stop();
+      provider?.removeAllListeners();
```

Comment:
I was going to bring this over to the extension side, but Mark [pointed out](https://github.com/MetaMask/metamask-extension/issues/18647) that we may not actually want to do this, because the provider proxy will transfer existing event listeners to the new provider when it replaces the old one. In any case, now that I'm looking at this again, it doesn't appear that this is equivalent to what `stop` was doing before — stopping the provider removed event listeners from the _block tracker_. We don't need to do that, either, because those should get transferred over to the new block tracker too. So it looks like we can remove `#safelyStopProvider`.